nextmv/nextmv/local/executor.py

```diff
@@ -0,0 +1,565 @@
+    if raw_output.strip() != "":
+        stdout_output = json.loads(raw_output)
+
+    # Create outputs directory.
+    outputs_dir = os.path.join(run_dir, OUTPUTS_KEY)
+    os.makedirs(outputs_dir, exist_ok=True)
+    temp_run_outputs_dir = os.path.join(temp_src, OUTPUTS_KEY)
+
+    process_run_information(
+        run_id=run_id,
+        run_dir=run_dir,
+        result=result,
+    )
+    process_run_logs(
+        run_dir=run_dir,
+        result=result,
+    )
+    process_run_statistics(
+        temp_run_outputs_dir=temp_run_outputs_dir,
+        outputs_dir=outputs_dir,
+        stdout_output=stdout_output,
+    )
+    process_run_assets(
+        temp_run_outputs_dir=temp_run_outputs_dir,
+        outputs_dir=outputs_dir,
+        stdout_output=stdout_output,
+    )
+    process_run_solutions(
+        run_id=run_id,
+        run_dir=run_dir,
+        temp_run_outputs_dir=temp_run_outputs_dir,
+        temp_src=temp_src,
+        outputs_dir=outputs_dir,
+        stdout_output=stdout_output,
+    )
+    process_run_visuals(
+        run_dir=run_dir,
+        outputs_dir=outputs_dir,
+    )
+
+
+def process_run_information(run_id: str, run_dir: str, result: subprocess.CompletedProcess[str]) -> None:
+    """
+    Processes the run information, updating properties such as duration and
+    status.
+
+    Parameters
+    ----------
+    run_id : str
+        The ID of the run.
+    run_dir : str
+        The path to the run directory.
+    result : subprocess.CompletedProcess[str]
+        The result of the subprocess run.
+    """
+
+    info_file = os.path.join(run_dir, f"{run_id}.json")
+
+    with open(info_file) as f:
+        info = json.load(f)
+
+    # Calculate duration.
+    created_at_str = info["metadata"]["created_at"]
+    created_at = datetime.fromisoformat(created_at_str.replace("Z", "+00:00"))
+    now = datetime.now(timezone.utc)
+    duration = round((now - created_at).total_seconds() * 1000, 1)
+
+    # Update the status
+    status = StatusV2.succeeded.value
+    error = ""
+    if result.returncode != 0:
+        status = StatusV2.failed.value
+        error = result.stderr if result.stderr else "unknown error"
+
+    # Update the run info file.
+    info["metadata"]["duration"] = duration
+    info["metadata"]["status_v2"] = status
+    info["metadata"]["error"] = error
+
+    with open(info_file, "w") as f:
+        json.dump(info, f, indent=2)
+
+
+def process_run_logs(run_dir: str, result: subprocess.CompletedProcess[str]) -> None:
+    """
+    Processes the logs of the run. Writes the logs to a logs directory.
+
+    Parameters
+    ----------
+    run_dir : str
+        The path to the run directory.
+    result : subprocess.CompletedProcess[str]
+        The result of the subprocess run.
+    """
+
+    logs_dir = os.path.join(run_dir, LOGS_KEY)
+    os.makedirs(logs_dir, exist_ok=True)
+    with open(os.path.join(logs_dir, LOGS_FILE), "w") as f:
+        f.write(result.stderr)
```

Comment:
I think we need a bit more complex stdout/stderr handling logic. For `multi-file` we consider stdout+stderr as the log-stream. So, we need to have them both merged into one log stream when capturing. I don't think we can do this post run, right? :thinking: